### PR TITLE
[C-2665 C-2714] Add developer apps settings modal

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Adds developer apps settings modal
- Shows users developer apps
- Allows user to create new app
- New app saved to current developer apps
- Each app can be viewed or deleted

### TODO
Hook up getDeveloperApps sdk method (will publish and pull in latest sdk)
Finalize last navigation items around viewing and deleting developer app

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2665-create-app-flow

File: packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  appName: string
+}
+
+export const DeveloperAppListItem = (props: DeveloperAppListItemProps) => {
+  const { index, appName } = props
+
+  const divider = <hr className={styles.listItemDivider} />
+
+  return (
+    <li className={styles.listItem}>
```
[the remark this answers is not in the document]
Comment (nicoback2) [reply packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppListItem.tsx:20]: nvm looks like the parent gives the key

File: packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppsSettingsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+import { ReactComponent as IconEmbed } from 'assets/img/iconEmbed.svg'
+
+import { AppDetailsPage } from './AppDetailsPage'
+import { CreateNewAppPage } from './CreateNewAppPage'
+import { YourAppsPage } from './YourAppsPage'
+import { CreateAppsPages } from './types'
+
+const messages = {
+  title: 'Your Apps',
```
Comment (nicoback2) [inline packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppsSettingsModal.tsx:19]: Looks like these are intentionally the same as the ones in YourAppsPage so prob better to share the same `messages` obj across the two files

File: packages/common/src/api/developer-apps.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import { z } from 'zod'
+
+import { ID } from 'models/Identifiers'
+import { createApi } from 'src/audius-query'
+import { encodeHashId } from 'utils/hashIds'
+
+const descriptionMaxLength = 28
```
Comment (nicoback2) [inline packages/common/src/api/developer-apps.ts:7]: i think this should be 128 (assuming it's chars)

File: packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppsSettingsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  return (
+    <>
+      <Modal {...props} size='small'>
+        <ModalHeader>
+          <ModalTitle title={messages.title} icon={<IconEmbed />} />
+        </ModalHeader>
+        <ModalContentPages currentPage={getCurrentPage(currentPage)}>
+          <YourAppsPage setPage={handleSetPage} />
+          <CreateNewAppPage setPage={handleSetPage} />
```
Comment (nicoback2) [inline packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppsSettingsModal.tsx:77]: Not sure if you were leaving this for next PR but this should be disabled + show tooltip if the user has already created 3 apps

File: packages/web/src/pages/settings-page/components/desktop/DeveloperApps/CreateNewAppPage.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  const { status, data } = result
+
+  useEffect(() => {
+    if (status === Status.SUCCESS && data) {
+      setPage(CreateAppsPages.APP_DETAILS, data)
+    }
+  })
+
+  const handleSubmit = useCallback(
+    (values: DeveloperAppValues) => {
```
Comment (nicoback2) [inline packages/web/src/pages/settings-page/components/desktop/DeveloperApps/CreateNewAppPage.tsx:49]: might want to check here if user has already created 3 apps and abort if so. but we can/should also do that in the sdk...

File: packages/common/src/api/developer-apps.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+export const developerAppSchema = z.object({
+  userId: z.number(),
+  name: z.string().min(5),
```
Comment (nicoback2) [inline packages/common/src/api/developer-apps.ts:11]: Not sure if name needs a min,  maybe run by Forrest?

**Review by nicoback2 — APPROVED** [review]
woo looks great, left some comments!

File: packages/web/src/pages/settings-page/components/desktop/DeveloperApps/CreateNewAppPage.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  descriptionLabel: 'Short Description',
+  cancel: 'Cancel',
+  create: 'Create Key'
+}
+
+type CreateNewAppPageProps = CreateAppPageProps
+
+export const CreateNewAppPage = (props: CreateNewAppPageProps) => {
+  const { setPage } = props
+  const userId = useSelector(getUserId) as number
```
[the remark this answers is not in the document]
Comment (dylanjeffers) [reply packages/web/src/pages/settings-page/components/desktop/DeveloperApps/CreateNewAppPage.tsx:36]: its nullable, i could just not have it and do checks

File: packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppListItem.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+type DeveloperAppListItemProps = {
+  index: number
+  appName: string
+}
+
+export const DeveloperAppListItem = (props: DeveloperAppListItemProps) => {
+  const { index, appName } = props
+
+  const divider = <hr className={styles.listItemDivider} />
```
[the remark this answers is not in the document]
Comment (dylanjeffers) [reply packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppListItem.tsx:17]: ya totes, i also need to determine if we should just use the special divider component, can look into

File: packages/common/src/api/developer-apps.ts (hunk 3 above)
Comment (dylanjeffers) [reply packages/common/src/api/developer-apps.ts:7]: oh yup great find

File: packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppsSettingsModal.tsx (hunk 4 above)
Comment (dylanjeffers) [reply packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppsSettingsModal.tsx:77]: have not addressed that yet, thanks for headsup

File: packages/web/src/pages/settings-page/components/desktop/DeveloperApps/CreateNewAppPage.tsx (hunk 5 above)
Comment (dylanjeffers) [reply packages/web/src/pages/settings-page/components/desktop/DeveloperApps/CreateNewAppPage.tsx:49]: i think as long as we hide the create button, should be good?

File: packages/common/src/api/developer-apps.ts (hunk 6 above)
Comment (dylanjeffers) [reply packages/common/src/api/developer-apps.ts:11]: word good point, ill remove the min anyway

**Pull request merged**